File: c-sharp/Controllers/Rot13Controller.cs
using System;
using System.Collections.Generic;
using System.Text;
using Ciphers.Models;

namespace Ciphers.Controllers
{
    public class Rot13Controller : BaseController
    {
        public Rot13Text GetRot13Text(Rot13Text rText)
        {
            foreach (char plnChar in rText.PlainText)
            {
                rText.StrBuild.Append(CipherService.GetCipherLetter(plnChar, rText.Key));
            }

            rText.CipherText = rText.StrBuild.ToString();
            return rText;
        }
    }
}

File: java/src/main/java/com/metatarsal/ciphers/controllers/Rot13Controller.java
package com.metatarsal.ciphers.controllers;

import com.metatarsal.ciphers.models.Rot13Text;
import org.springframework.stereotype.Controller;

@Controller
public class Rot13Controller extends BaseController {

    public Rot13Text rot13Cipher(Rot13Text text) {

        StringBuilder sb = getStringBuilder();

        for(int i = 0, n = text.getPlainText().length(); i < n; i++) {
            sb.append(cipherService.getCipherLetter(text.getPlainText().charAt(i),
                                                    text.getKEY()));
        }

        text.setCipherText(sb.toString());

        return text;
    }
}
